### ISSUE TITLE
`TerminalTk` "SanityCheck" error

### ISSUE THREAD

**Issue by TheLizzard** [TheLizzard/CPP-IDLE#3]
```python
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python3.10/tkinter/__init__.py", line 1921, in __call__
    return self.func(*args)
  File "/usr/lib/python3.10/tkinter/__init__.py", line 839, in callit
    func(*args)
  File "/path/src/bettertk/terminaltk/terminaltk.py", line 301, in handle_msg_loop
    self.handle_msg(*self.terminal.buffer.msg_queue.pop())
  File "/path/src/bettertk/terminaltk/terminaltk.py", line 336, in handle_msg
    assert self.running_state[0] == args[0], "SanityCheck"
AssertionError: SanityCheck
```
First F5, no cwd set.
The problem was probably when running `"mvn compile"`.

**Comment by TheLizzard** [comment]
#4 Isn't connected to this bug. Added a logger to the pipe and it should dump all of the data sent/received before it raises that `AssertionError`.

**Comment by TheLizzard** [comment]
```python
PipeLog:
	Read: b'STARTED'
	Wrote: b'PRINTSTR=================================== Starting ===================================\n\x00'
	Wrote: b'RUN\x00\x01\x00\x02cd\x00/home/thelizzard\x00\x00'
	Read: b'RUNNING\x00\x01'
	Read: b'EXITCODE\x00\x01\x00\x00\x00\x00'
Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python3.10/tkinter/__init__.py", line 1921, in __call__
    return self.func(*args)
  File "/usr/lib/python3.10/tkinter/__init__.py", line 839, in callit
    func(*args)
  File "/path/Bismuth-184/src/bettertk/terminaltk/terminaltk.py", line 301, in handle_msg_loop
    self.handle_msg(*self.terminal.buffer.msg_queue.pop())
  File "/path/Bismuth-184/src/bettertk/terminaltk/terminaltk.py", line 338, in handle_msg
    assert self.running_state[0] == args[0], "SanityCheck"
AssertionError: SanityCheck
```

Which decoded means:
```python
started
>>> print(...)
>>> run(cd, cmd_id=1, print_str="")
running cmd_id=1
exited cmd_id=1 exitcode=0
```

Therefore, the problem is a race condition somewhere in `slave.py`, `terminal.py`, or `terminaltk.py` (ranked from least to most likely).

PS, the run signal takes in a string to print out so I can I remove the `PRINTSTR` signal.

**Comment by TheLizzard** [comment]
`terminaltk.py`: `msg_queue.pop()` => `msg_queue.pop(0)` in 2 places. God this is annoying.
Will be fixed in next commit.

**Comment by TheLizzard** [comment]
Solved in Version 2.7
